fix: change git credential mounting docs to be devworkspace-specific (#2275)
eclipse-che/che-docs: modules/end-user-guide/pages/git-credential-store.adoc

:navtitle: Creating a {orch-name} Secret for a Git credentials store
:keywords: user-guide, configuring, user, secrets
:page-aliases: 

[id="mounting-a-git-credential-store-into-workspace-containers_{context}"]
= Creating a {orch-name} Secret for a Git credentials store

This section describes how to create a Secret containing a Git credentials store in the user's {orch-namespace}. The Git credentials store specified in the Secret is mounted as a file in the {devworkspace}'s containers.

.Prerequisites

* A running instance of {prod-short}.

> * You have CLI or GUI access to the {platforms-name} cluster of your organization's {prod-short} instance.

Reviewer: ```suggestion
* You must/should have CLI or GUI access to the {platforms-name} cluster of your organization's {prod-short} instance.
```

Reviewer 2: `must / should` really? please use one suggestion

.Procedure

> . Prepare a Git credential file in the link:https://git-scm.com/docs/git-credential-store#_storage_format[Storage format].

Reviewer 3: @dkwon17, I suggested the `[TIP]` based on some third-party how-to article online when I was trying to figure out with Google what https://git-scm.com/docs/git-credential-store meant. Have a second look, it doesn't seem to be there. This simplifies things (like looking up tokens) for users who have local Git repo clones.

> . Encode content of the file to the base64 format.

Reviewer 3: ```suggestion
. Copy the content of the `.git-credentials` file and encode it to Base64.
+
For CLI users: Paste the content of the `.git-credentials` file into the following commands to encode it to Base64:
+
[source,subs="+quotes,+attributes,+macros"]
----
$ echo -n '__<content_of_.git-credentials>__' | base64
----
```

> . Create a new {platforms-name} secret in the user's {orch-namespace}.

Reviewer: ```suggestion
. Create a new {platforms-name} Secret in the user's {orch-namespace}.
```

+
[source,yaml,subs="+quotes,+attributes,+macros"]
----
apiVersion: v1
kind: Secret
metadata:
  name: git-credentials-secret
  labels:
    controller.devfile.io/git-credential: 'true'                 <1>
    controller.devfile.io/watch-secret: 'true'
  annotations:
    controller.devfile.io/mount-path: __<path_in_container>__        <2>
data:

>   credentials: __<base64_content_of_the_git_credentials_store>__     <3>

Reviewer 3: David, if you decide to apply this, the number of spaces before <3> has changed and most likely needs to be corrected:
```suggestion
  credentials: __<Base64_content_of_.git-credentials>__     <3>
```

----
+

> <1> The `controller.devfile.io/git-credential` label marks the secret as containing Git credentials. All Git credential secrets in the same {orch-namespace} will be merged into a single secret that would be mounted to the {devworkspace} containers. All the original secrets are kept in-tact.

Reviewer: ```suggestion
<1> The `controller.devfile.io/git-credential` label marks the Secret because it contains the Git credentials. All Git credential Secrets in the same {orch-namespace} will be merged into a single Secret that would be mounted to the {devworkspace} containers. All the original Secrets are kept in-tact.
```

<2> The path in the {devworkspace}'s containers at which the merged Git credentials secrets are to be mounted. For example, if set to `/tmp/git-credentials`, the merged Git credentials store is mounted at `/tmp/git-credentials/credentials`.
<3> The Git credentials store content encoded with `base64`.

. Apply the secret by entering it with `{orch-cli}` on a command line:
+

> [source,subs="+quotes,+attributes,+macros"]

Reviewer 3: ```suggestion
[subs="+quotes,+attributes,+macros"]
```

endif::[]

ifeval::["{project-context}" == "crw"]
** For CLI users: `{orch-cli}` is installed in the operating system you are using. See link:https://docs.openshift.com/container-platform/4.10/cli_reference/openshift_cli/getting-started-cli.html#installing-openshift-cli[Installing the OpenShift CLI].
endif::[]

* For CLI users: link:https://www.gnu.org/software/coreutils/base64[`base64`] is installed in the operating system you are using.

.Procedure
. In your home directory, locate and open your `.git-credentials` file if you already have it. Alternatively, if you do not have this file, then save a new `.git-credentials` file, using the following link:https://git-scm.com/docs/git-credential-store#_storage_format[Git credentials storage format]. Each credential is stored on its own line in the file:

+
[subs="+quotes,+attributes,+macros"]
----
https://__<username>__:__<token>__@__<git_server_hostname>__
----

+
.A line in a `.git-credentials` file
====
`https://trailblazer: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@github.com`
====

> . Select credentials from your `.git-credentials` file to provide to the Secret. Encode the selected credentials to Base64 for the next step.

Reviewer 3: ```suggestion
. Select credentials from your `.git-credentials` file to provide to the Secret. Encode the selected credentials to Base64 for the next step.
+
[TIP]
====
CLI users can create a Base64-encoded string by running `base64`:
* To encode all lines in the file:
** `$ cat .git-credentials | base64 | tr -d '\n'`
* To encode a selected line:
** `$ echo -n '__<copied_and_pasted_line_from_.git-credentials>__' | base64`
====
```

:navtitle: Creating a {orch-name} Secret for a Git credentials store
:keywords: user-guide, configuring, user, secrets
:page-aliases: 

[id="mounting-a-git-credential-store-into-workspace-containers_{context}"]
= Creating a {orch-name} Secret for a Git credentials store

> This section describes how to create a Secret containing a Git credentials store in the user's {orch-namespace}. The Git credentials store specified in the Secret is mounted as a file in the {devworkspace}'s containers.

Reviewer 3: Also, I've removed the last sentence `This Secret is mounted as a file in the {devworkspace} containers. ` because this info seems to be restated in the next paragraph.

> When mounting the Secret, a Git configuration file with the path to the mounted Git credentials store is automatically configured and mounted to the {devworkspace} containers at `/etc/gitconfig`.

Reviewer 3: ```suggestion
When mounting the Secret, a Git configuration file with the path to the mounted Git credentials store is automatically configured and mounted to the {devworkspace} containers at `/etc/gitconfig`.
```
